obspy/taup/tau.py

```diff
@@ -346,17 +346,15 @@ def plot_rays(self, phase_list=None, plot_type="spherical",
 
             intp = matplotlib.cbook.simple_linear_interpolation
             radius = self.model.radius_of_planet
+            phase_names_encountered = set([ray.name for ray in arrivals])
+            colors = {name: COLORS[i % len(COLORS)]
+                      for i, name in enumerate(phase_names_encountered)}
```

Comment:
This probably won't be sorted, so it might not be consistent.

Reply:
Good catch